### DISCUSSION TITLE
How to catch ALL exceptions in Azure Function v4

### DISCUSSION THREAD

**Discussion by itimpo** [Azure/azure-functions-dotnet-worker#2437] [Feedback]
Hi,

I use Microsoft.Azure.Functions.Worker v1.21.0
And I need to change response for standard exceptions like 404
When Azure Function doesn't have requested http method for path it just returns 404, and I need to add custom response message.

So, the question is How can I catch and handle all Exceptions BEFORE standard handler

I have ExceptionMiddleware and I use it like this
```
 .ConfigureFunctionsWebApplication(worker =>
 {
    worker.UseMiddleware<ExceptionMiddleware>();
})
```

But it doesn't catch low level exceptions

Thanks and Best Regards,
Dmitry

**Comment by artmasa** [comment]
I think this would be very hard. Because the host only forwards requests that match your functions to the worker as GRPC requests. Everything else is only handled by the host and never make their way to the worker process.
